### PR TITLE
Feature/defeat detection

### LINKED ISSUE #11: Detect when a player is defeated
Description:
As a developer, I want to determine when a player is defeated.

Acceptance Criteria:
- If HP <= 0 → creature is dead
- Battle result includes winner
- No further turns allowed

### PR DESCRIPTION
**Pull request by Typhon3**
closes #11

### CODE REVIEW

**Review by Soundscape93 — CHANGES_REQUESTED** [review]
please address the inline comments

File: test/MysticRiver.UnitTests/MysticRiver.UnitTests.csproj [hunk 1]
```diff
@@ -14,8 +14,15 @@
...
+  </ItemGroup>
+
   <ItemGroup>
     <Using Include="Xunit" />
   </ItemGroup>
 
 </Project>
+
+
+</Project>
```
[the remark this answers is not in the document]
Comment (Soundscape93) [reply test/MysticRiver.UnitTests/MysticRiver.UnitTests.csproj]: resolved with merge conflicts

File: test/MysticRiver.UnitTests/DefeatDetectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    {
+        var creature = new Creature("Gruk", 100);
+
+        creature.TakeDamage(30);
+
+        Assert.Equal(70, creature.CurrentHp);
+    }
+}
+
+public class BattleTests
```
Comment (Soundscape93) [inline test/MysticRiver.UnitTests/DefeatDetectionTests.cs:65]: Could we add tests for Battle constructor guards: null player1, null player2, and same creature for both players should throw?

**Review by Soundscape93 — APPROVED** [review]
